specs/Pills.md

```diff
@@ -0,0 +1,188 @@
+# [Pills]()
+
+## Background
+
+PIlls should be used when representing an input, as a way to filter content, or to represent an attribute.
+
+## Prior Art
+
+- [Evergreen](https://evergreen.segment.com/components/badge-and-pill/)
+- [Lightning](https://www.lightningdesignsystem.com/components/pills/)
+- [OpenUI Research]()
+
+## Sample Code
+
+```jsx
+<Pills>
+  <Pill>Item 1</Pill>
+  <Pill>Item 2</Pill>
+</Pills>
+```
+
+## Variants
+
+- Shape: `circular`, `rounded`
+- Size: `smaller`, `small`, `medium`
+- Appearance: `Filled Grey (filled)`, `Filled White (inverted)`, `Outline`
+
+## PROPS
+
+```typescript
+export type PillShape = 'rounded' | 'circular';
+
+export type PillAppearence = 'filled' | 'inverted' | 'outline';
+
+export type SizeValue = 'smaller' | 'small' | 'medium';
+
+export interface PillProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * A Pill can be sized.
+   */
+  size?: SizeValue;
+
+  /**
+   * A Pill can be circular or rounded
+   */
+  shape?: PillShape;
+
+  /**
+   * A Pill can be filled, inverted or outline
+   */
+  appearance?: PillVariant;
+
+  /**
+   * Media slot
+   */
+  media?: ShorthandProps<HTMLElement>;
+
+  /**
+   * Details that will appear as second line
+   */
+  details?: ShorthandProps<HTMLElement>;
+
+  /** A Pill can be dismissible. */
+  dismissible?: boolean;
+
+  /** A Pill can be disabled. */
+  disabled?: boolean;
+
+  /** A Pill can be selected. */
+  selected?: boolean;
+
+  /** A Pill can be draggable  */
+  draggable?: boolean;
+
+  /**
+   * A button shorthand for the dismiss action slot. To use this slot the pill should be
+   * dismissible.
+   */
+  dismissAction?: ShorthandValue<AlertDismissActionProps>;
+}
+```
+
+## Structure
+
+- _**Public**_
+
+```jsx
+<Pills>
+  <Pill />
+</Pills>
+```
+
+- _**Internal**_
+
+```jsx
+<ElementType >
+  <li>
+    <span>
+      {children}
+    </span>
+  <li>
+</ElementType>
+```
+
+- _**DOM**_
+
+```html
+<ul aria-label>
+  <li role="presentation">
```

Comment:
I've never understood the value of having these `li` wrappers with `role ="presentation"`, it's purely to make sure that the usage with `ul` is correct. 

In this case doesn't it make more sense to use divs ?

Reply:
Well, I guess just to be semantically correct.

Reply:
but why not just use `<li role="button">` ? 

presentation just kills the default list role anyway